### PR TITLE
Rsarkar/communication chat preview5

### PR DESCRIPTION
**Pull request by sarkar-rajarshi**

### CODE REVIEW

File: sdk/communication/azure-communication-chat/README.md [hunk 1]
```diff
@@ -115,6 +114,11 @@ Perform Update operation on thread topic
 update_topic(topic, **kwargs)
 ```
 
+## Get Chat thread properties
+```python
+get_properties(**kwargs)
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/README.md:119]: The snippets will need a bit more context than this (and this applies to other existing snippets in the readme).
The goal is that a user should be able to copy+paste the code and use it right away.
So something more like:
```python
client = ChatClient(...)
thread = client.get_chat_thread_client(...)
thread.get_properties(...)
```

Given that this applies to multiple snippets throughout the readme - I'm happy if you want to open a github issue to address it in a subsequent PR.

File: sdk/communication/azure-communication-chat/README.md [hunk 2]
```diff
@@ -240,16 +243,17 @@ def decide_to_retry(error, **kwargs):
     return True
 
 retry = [thread_participant for thread_participant, error in create_chat_thread_result.errors if decide_to_retry(error)]
-chat_thread_client.add_participants(retry)
+if len(retry) > 0:
+    chat_thread_client.add_participants(retry)
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/README.md:247]: `add_participants` should be a no-op if passed an empty list, so a user should be able to do:
```python
chat_thread_client.add_participants([p for p, error in create_chat_thread_result.errors if decide_to_retry(error)])
```
Additionally, if you need to check that a list has 0 elements, you can just do this in python:
```python
need_to_retry = [p for p, error in create_chat_thread_result.errors if decide_to_retry(error)]
if need_to_retry:
    chat_thread_client.add_participants(need_to_retry)
```

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py [hunk 3]
```diff
@@ -117,6 +119,32 @@ def thread_id(self):
         """
         return self._thread_id
 
+    @distributed_trace
+    def get_properties(
+        self,
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> ChatThreadProperties
+        """Gets a chat thread properties.
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:128]: Maybe "Gets the properties of the chat thread."?

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py [hunk 4]
```diff
@@ -117,6 +119,32 @@ def thread_id(self):
...
+
+        :return: ChatThreadProperties
+        :rtype: ~azure.communication.chat.ChatThreadProperties
+        :raises: ~azure.core.exceptions.HttpResponseError, ValueError
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:132]: Under what circumstances is `ValueError` raised? Unless it's something specific that users need to be aware of and code for, we can just remove this.

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py [hunk 5]
```diff
@@ -467,76 +495,25 @@ def list_participants(
...
-                               communication_error.message)
 
     @distributed_trace
     def add_participants(
         self,
         thread_participants,  # type: list[ChatThreadParticipant]
         **kwargs  # type: Any
     ):
-        # type: (...) -> list[(ChatThreadParticipant, CommunicationError)]
+        # type: (...) -> list[(ChatThreadParticipant, ChatError)]
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:505]: I think the typehint should be something like:
```python
from typing import List, Tuple
List[Tuple[ChatThreadParticipant, ChatError]]
```

Comment (annatisch) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py:505]: I see this is what you already had in the docstring `:return:` value :)

File: sdk/communication/azure-communication-chat/azure/communication/chat/_generated/operations/_chat_operations.py [hunk 6]
```diff
@@ -127,7 +126,7 @@ def list_chat_threads(
         start_time=None,  # type: Optional[datetime.datetime]
         **kwargs  # type: Any
     ):
-        # type: (...) -> Iterable["_models.ChatThreadsInfoCollection"]
+        # type: (...) -> Iterable["_models.ChatThreadsItemCollection"]
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/azure/communication/chat/_generated/operations/_chat_operations.py:129]: I'm not sure if this is a bug in autorest or swagger spec - but I think this should be returning an `Iterable["_models.ChatThreadItem"]`.
It doesn't really matter in the generated code, so long as we've confirmed that the hand-written layer over the top is doing the right thing :)

File: sdk/communication/azure-communication-chat/samples/chat_thread_client_sample.py [hunk 7]
```diff
@@ -84,20 +84,37 @@ def create_chat_thread_client(self):
...
         print("chat_thread_client created")
 
+    def get_chat_thread_properties(self):
+        thread_id = self._thread_id
+        # [START get_thread]
+        from azure.communication.chat import ChatClient
+        from azure.communication.identity._shared.user_credential import CommunicationTokenCredential
+        from azure.communication.chat._shared.user_token_refresh_options import CommunicationTokenRefreshOptions
+
+        refresh_options = CommunicationTokenRefreshOptions(self.token)
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/samples/chat_thread_client_sample.py:94]: I believe we refactored out this `CommunicationTokenRefreshOptions` object

File: sdk/communication/azure-communication-chat/samples/chat_thread_client_sample_async.py [hunk 8]
```diff
@@ -80,24 +80,46 @@ async def create_chat_thread_client_async(self):
         print("thread created, id: " + self._thread_id)
         print("create_chat_thread_client_async succeeded")
 
+    async def get_chat_thread_properties_async(self):
+        thread_id = self._thread_id
+        # [START get_thread]
+        from azure.communication.chat.aio import ChatClient
+        from azure.communication.identity._shared.user_credential_async import CommunicationTokenCredential
+        from azure.communication.chat._shared.user_token_refresh_options import CommunicationTokenRefreshOptions
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/samples/chat_thread_client_sample_async.py:88]: The credentials objects should just be imported from `azure.communication.chat` now right?

File: sdk/communication/azure-communication-chat/tests/test_chat_thread_client_e2e_async.py [hunk 9]
```diff
@@ -406,3 +384,17 @@ async def test_list_read_receipts(self):
 
             if not self.is_playback():
                 await self.chat_client.delete_chat_thread(self.thread_id)
+
+    @pytest.mark.live_test_only
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/tests/test_chat_thread_client_e2e_async.py:388]: Why is this marked live test only?

File: sdk/communication/azure-communication-chat/tests/test_chat_thread_client_async.py [hunk 10]
```diff
@@ -735,3 +679,27 @@ async def mock_send(*_, **__):
...
 
     assert len(items) == 1
+
+@pytest.mark.asyncio
+async def test_get_properties():
+    thread_id = "19:bcaebfba0d314c2aa3e920d38fa3df08@thread.v2"
+    raised = False
+
+    async def mock_send(*_, **__):
+        return mock_response(status_code=200, json_payload={
```
Comment (annatisch) [inline sdk/communication/azure-communication-chat/tests/test_chat_thread_client_async.py:689]: Why do we need to use mock responses here? Is there some reason the recording wont work?

**Review by juancamilor — APPROVED** [review]

File: sdk/communication/azure-communication-chat/README.md (hunk 1 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/README.md:116]: Will update the README and the samples in a separate PR

File: sdk/communication/azure-communication-chat/README.md (hunk 2 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/README.md:244]: `add_participants` has been updated to be no-op if passed empty list. Will update the README and the samples in a separate PR.

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py (hunk 3 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py]: Changed

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py (hunk 4 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py]: It was a miss. It does not throw ValueError in any scenario. This sneaked in when transferring the method from chat_client :)

File: sdk/communication/azure-communication-chat/azure/communication/chat/_generated/operations/_chat_operations.py (hunk 6 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_generated/operations/_chat_operations.py:129]: This is an autorest thing. It is exposed as `ItemPaged[ChatThreadItem]` to the outside world.

File: sdk/communication/azure-communication-chat/samples/chat_thread_client_sample.py (hunk 7 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/samples/chat_thread_client_sample.py]: yes. The changes were in a dangling in a separate PR #17178 . It's been merged.

File: sdk/communication/azure-communication-chat/samples/chat_thread_client_sample_async.py (hunk 8 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/samples/chat_thread_client_sample_async.py]: yes. The changes were in a dangling in a separate PR #17178 . It's been merged.

File: sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py (hunk 5 above)
Comment (sarkar-rajarshi) [reply sdk/communication/azure-communication-chat/azure/communication/chat/_chat_thread_client.py]: It used to show up as a flag in apiview before. Have refactored this

**Pull request merged**